api/server/httputils/errors.go

```diff
@@ -54,6 +54,7 @@ func GetHTTPErrorStatusCode(err error) int {
 			code    int
 		}{
 			{"not found", http.StatusNotFound},
+			{"cannot find", http.StatusNotFound},
```

Comment:
Sure, I'd be happy to.

This code that you've added to is trying to guess which http status code to return based on an error message string. Instead of guessing based on string matching, we'd like to be explicit about the error.

The explicit error is in `api/errors/errors.go` `NewRequestNotFoundError()`. Could you return that directly from the function that causes this error, instead of adding a new string to this list?